torchgeo/datasets/nasa_marine_debris.py

```diff
@@ -99,6 +99,12 @@ def __getitem__(self, index: int) -> Dict[str, Tensor]:
         boxes = self._load_target(self.files[index]["target"])
         sample = {"image": image, "boxes": boxes}
 
+        # Filter invalid boxes
+        w_check = (sample["boxes"][:, 2] - sample["boxes"][:, 0]) > 0
+        h_check = (sample["boxes"][:, 3] - sample["boxes"][:, 1]) > 0
+        indices = w_check & h_check
+        sample["boxes"] = sample["boxes"][indices]
```

Comment:
How many of the boxes were invalid?

@isaaccorley recently had to do the same thing for IDTReeS, not sure if we should try to reuse code for that.

Reply:
We can reuse the IDTReeS._filter_boxes method and make it a function so it can be used for both datasets. Doesn't have to be in this PR but it seems as though this may be a common issue with object detection datasets.